torch_xla/csrc/xla_sharding_util.cpp

```diff
@@ -488,8 +489,11 @@ std::vector<at::Tensor> ShardingUtil::ShardTensor(
             shards[i], c10::IntArrayRef(pads.data(), pads.size()), 0);
       }
     }
-  } else if ((sharding.type() == xla::OpSharding::MANUAL) ||
-             (sharding.type() == xla::OpSharding::TUPLE)) {
+  } else if ((sharding.type() == xla::OpSharding::MANUAL)) {
+    // Just put the full tensor on the first device.
+    shards[0] = tensor;
+    shards.resize(1);
```

Comment:
How does this work for a compuatation, since we need to feed each device some input data?

e.g. based on your unit test, what happens if we run:

```python
    x = torch.randn(3, 2)
    xx = x.to(xm.xla_device())  # xx is device data
    xt = xs._mark_manual_sharding(xx)

    ones = torch.ones(3, 2).to(xm.xla_device()) # ones is replicated to all devices
    print(xx + ones)  # What will happen here?
```

Reply:
XLA should assume that `xt` is sharded manually, so expected to be plicated as well. The purpose of `MANUAL` is to support custom kernel and prevent XLA to override the manual sharding.

Reply:
Good question. I would expect it behaves as a single device. Let me double check as well.